### DISCUSSION TITLE
Bem vindo a discussões em hbsefazclass

### DISCUSSION THREAD

**Discussion by JoseQuintas** [JoseQuintas/sefazclass#14] [General]
## 👋 Bem vindo!

Testando o uso de discussões diretamente no repositório do github

**Comment by MarceloBrigatti** [comment]
Olá José

Quanto a nova exigência da Tag xPag da Forma de pagamento em detPag quando for utilizado "99", posso sugerir essa alteração em ze_sefazdanfe.prg apartir da linha 673 

      FOR EACH cTPag IN ::aDetPag
         cNumero := XmlNode( cTPag, "tPag" )
         cValor  := AllTrim( FormatNumber( Val( XmlNode( cTPag, "vPag" ) ), 13, 2 ) )
         cDescr  := AllTrim( XmlNode( cTPag, "xPag" ) )
         IF nColuna > 3
            ::nLinhaPdf -= 8
            nColuna := 1
         ENDIF
         IF ( nPos := hb_AScan( aList, { | e | e[ 1 ] == cNumero } ) ) != 0
            If nPos == Len( aList ) 
               ::DrawTexto( 6 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), ::nLinhaPdf - 1, 122 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), NIL, cDescr, HPDF_TALIGN_LEFT, ::oPDFFontNormal, 8 )
            Else
               ::DrawTexto( 6 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), ::nLinhaPdf - 1, 122 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), NIL, aList[ nPos, 2 ], HPDF_TALIGN_LEFT, ::oPDFFontNormal, 8 )
            EndIf
         ENDIF
         ::DrawTexto( 130 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), ::nLinhaPdf - 1, 195 + ( ( ( nTamForm ) / 3 ) * ( nColuna - 1 ) ), NIL, cValor, HPDF_TALIGN_RIGHT, ::oPDFFontNormal, 8 )
         nColuna++
      NEXT

**Comment by JoseQuintas** [comment]
de onde tirou a tag xPag ?

**Comment by MarceloBrigatti** [comment]
Boa tarde.


Alterações da versão 1.20

A versão 1.20 da nota técnica 2020.006 publicada em março de 2021
apresentou muitas mudanças nas regras de validação, abaixo vamos falar de
cada uma delas; o prazo da versão ficou assim definido: homologação em
03/05/2021 e produção em 01/09/2021.

   - Grupo YA. Informações de Pagamento (Leiaute da Nota Fiscal Eletrônica)
   YA02a tag: xPag – Criado o campo Descrição do Meio de Pagamento (xPag)
   para preenchimento do meio de pagamento quando for utilizado o código do
   meio de pagamento 99-outros


Dessa alteração. Senão dá essa rejeição:
Rejeição 441: Descrição do pagamento obrigatória para meio de pagamento 99-
outros

--
Sem mais, cordialmente,




Em qua., 1 de nov. de 2023 às 17:03, José M. C. Quintas <
***@***.***> escreveu:

**Comment by JoseQuintas** [comment]
Ok, confirmado.
No manual não consta, mas nessa nota técnica sim
Alterando aqui:
iif( cNumero == "99", XmlNode( cTPag, "xPag" ), aList[ nPos, 2 ] )
